examples/sphere/opt_limiters_solidbody.jl

```diff
@@ -0,0 +1,311 @@
+                end
+            end
+            if rd[1] < r0 &&
+               abs(λ - centers[1].long) >= r0 / 6 &&
+               ϕ - centers[1].lat < -5 * r0 / 12
+                q = 1.0
+            elseif rd[2] < r0 &&
+                   abs(λ - centers[2].long) >= r0 / 6 &&
+                   ϕ - centers[2].lat > 5 * r0 / 12
+                q = 1.0
+            else
+                q = 0.1
+            end
+        elseif test_name == gaussian_test_name
+            q = 0.95 * (exp(-5.0 * (rd[1] / r0)^2) + exp(-5.0 * (rd[2] / r0)^2))
+        else # default test case, cosine bells
+            for i in length(rd)
+                qi = Vector{Float64}(undef, 2)
+                if rd[i] < r0
+                    qi[i] = 1 / 2 * (1 + cospi(rd[i] / r0))
+                    q = 0.1 + 0.9 * qi[i]
+                else
+                    qi[i] = 0.0
+                    q = 0.1
+                end
+            end
+        end
+
+        # Initialize density
+        ρ = 1.0
+
+        return (ρ = ρ, ρq = ρ * q)
+    end
+
+    function f!(ystar, y, parameters, t)
+
+        grad = Operators.Gradient()
+        wdiv = Operators.WeakDivergence()
+        T = parameters.T
+
+        coords = Fields.coordinate_field(axes(y.ρ))
+        u = map(coords) do coord
+            ϕ = coord.lat
+            λ = coord.long
+
+            uu =
+                k * sind(λ)^2 * sind(2 * ϕ) * cospi(t / T) + 360.0 * cosd(ϕ) / T
+            uv = k * sind(2λ) * cosd(ϕ) * cospi(t / T)
+            Geometry.UVVector(uu, uv)
+        end
+
+        # Compute min_Q[] and max_Q[] that will be needed later in the stage limiter
+        space = parameters.space
+        n_elems = Topologies.nlocalelems(space)
+        topology = space.topology
+
+        neigh_elems_Q_min = Array{Float64}(undef, 8)
+        neigh_elems_Q_max = Array{Float64}(undef, 8)
+
+        for e in 1:n_elems
+            Q_e = Fields.slab(y.ρq, e)
+
+            Q_e_min = minimum(Q_e)
+            Q_e_max = maximum(Q_e)
+            neigh_elems = Topologies.neighboring_elements(topology, e)
+            for i in 1:length(neigh_elems)
+                if neigh_elems[i] == 0
+                    neigh_elems_Q_min[i] = +Inf
+                    neigh_elems_Q_max[i] = -Inf
+                else
+                    neigh_elems_Q_min[i] =
+                        Fields.minimum(Fields.slab(y.ρq, neigh_elems[i]))
+                    neigh_elems_Q_max[i] =
+                        Fields.maximum(Fields.slab(y.ρq, neigh_elems[i]))
+                end
+            end
+            min_Q[e] = min(minimum(neigh_elems_Q_min), Q_e_min)
+            max_Q[e] = max(maximum(neigh_elems_Q_max), Q_e_max)
+            parameters.min_Q[][e] = min_Q[e]
+            parameters.max_Q[][e] = max_Q[e]
+        end
+
+        # Compute hyperviscosity for the tracers equation by splitting it in two diffusion calls
+        @. ystar.ρq = wdiv(grad(y.ρq))
+        @. ystar.ρq = -D₄ * wdiv(grad(ystar.ρq))
```

Comment:
@OsKnoth yes, we only apply hyperdiffusion to the `ρq` variable, not to `ρ` alone. This is equation (16) in the reference paper Guba et al JCP 2014

Reply:
I mean it in the following form.
ystar.ρq = wdiv(grad(y.ρq/y.ρ))
ystar.ρq = -D₄ * wdiv(y.ρ*grad(ystar.ρq))

Reply:
That's not how it's presented in the paper. Though I can try. I don't think it will make a difference in this case, since `ρ=1` at t=0 and it should be preserved